### PR TITLE
fixing radio prop updates

### PR DESCRIPTION
**Pull request by wdrescher**
Adding memoization to the radio -> form group control pipeline. Now able to update the value of the radio group by adding by changing `selectedValue` on the radiogroup prop. Also added `values` to output of radio handler function to align with docs

### CODE REVIEW

File: src/components/form-controls/radio/RadioGroup.jsx [hunk 1]
```diff
@@ -32,10 +32,12 @@ const RadioGroup = ({
   handleChange,
   label,
   name,
-  selectedValue,
+  selectedValue: selectedVal,
   style,
   required,
 }) => {
+  const selectedValue = useMemo(() => selectedVal, [selectedVal]);
```
Comment (warrensplayer) [inline src/components/form-controls/radio/RadioGroup.jsx:39]: Do not use `useMemo` here.  This whole file could be reverted.

File: src/components/form-controls/radio/Radio.jsx [hunk 2]
```diff
@@ -45,7 +45,8 @@ const Radio = ({
   return (
     <div className={classes} style={style}>
       <input
-        defaultChecked={defaultChecked}
+        checked={checked}
+        defaultChecked={checked}
```
Comment (warrensplayer) [inline src/components/form-controls/radio/Radio.jsx:49]: Only set `checked` here.  You can remove the `defaultChecked` property

File: src/components/form-controls/core/form-control-group/FormControlGroup.jsx [hunk 3]
```diff
@@ -102,7 +102,7 @@ const FormControlGroup = (
 
     updateValidatorState(value);
     setFieldValues([value]);
-    handler(event);
+    handler(event, [value]);
```
Comment (warrensplayer) [inline src/components/form-controls/core/form-control-group/FormControlGroup.jsx:105]: I do not believe this change actually does anything unless the handler function was expecting a second parameter.

File: src/components/form-controls/checkbox/CheckboxGroup.jsx [hunk 4]
```diff
@@ -51,6 +51,8 @@ const CheckboxGroup = ({
     className,
   );
 
+  const selectedValues = useMemo(() => selectedVal, [selectedVal]);
```
Comment (warrensplayer) [inline src/components/form-controls/checkbox/CheckboxGroup.jsx:54]: Don't use `useMemo` here.  This whole file could be reverted.

**Review by warrensplayer — CHANGES_REQUESTED** [review]

File: src/components/form-controls/core/form-control-group/FormControlGroup.jsx (hunk 3 above)
Comment (wdrescher) [reply src/components/form-controls/core/form-control-group/FormControlGroup.jsx:105]: Yeah the handler function in the docs expects `event` and `values`. I think that aligning with the checkbox group is useful to keep things consistent between the two, but open to removing the `values` and updating the docs

**Pull request merged**